istioctl/pkg/injector/injector-list_test.go

```diff
@@ -105,3 +105,43 @@ func Test_getNamespaces(t *testing.T) {
 		assert.Equal(t, true, expected.Contains(ns.Name))
 	}
 }
+
+func Test_injectionDisabled(t *testing.T) {
+	cases := []struct {
+		name     string
+		pod      *corev1.Pod
+		expected bool
+	}{
+		{
+			name: "Injection enabled by annotation",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotation.SidecarInject.Name: "true",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Injection enabled by label",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotation.SidecarInject.Name: "false",
+					},
+					Labels: map[string]string{
+						label.SidecarInject.Name: "true",
+					},
+				},
+			},
+			expected: false,
+		},
+	}
+
+	for i, c := range cases {
+		t.Run(fmt.Sprintf("case %d %s", i, c.name), func(t *testing.T) {
+			assert.Equal(t, c.expected, injectionDisabled(c.pod))
```

Comment:
do you want to test disabled, why these two cases are both `enabled`

Reply:
> do you want to test disabled, why these two cases are both `enabled`

hi @hzxuzhonghu,I previously included all the testcases with enabled and disabled both which includes all the conditions,but they were providing the same code coverage as these two only so removed.
Do I need to cover all the testcases?

Reply:
We donot only care about coverage reported, we should look at the corner cases it coverred

Reply:
yes @hzxuzhonghu,agree with you,PTAL 7a5f0ff4d4a45e3e1622ad680c3444bcd91d5f3f,I have tried to cover most of the testcases,do I need to add more,if not,I will revert the changes.

Reply:
> We donot only care about coverage reported, we should look at the corner cases it coverred

Agree, this is why I wish `go test` wasn't so crude in terms of coverage options and supported `branch` rather than `statement` coverage - branch coverage _is_ a good proxy for "corner cases".

In this case, we are adding simple unit tests to a simple equality function with no subtle behavior that needs to be fenced, so statement coverage is "good enough".